src/lints/enum_unmarked_non_exhaustive.ron

```diff
@@ -0,0 +1,52 @@
+SemverQuery(
+    id: "enum_unmarked_non_exhaustive",
+    human_readable_name: "enum unmarked #[non_exhaustive]",
+    description: "A Public API enum is no longer #[non_exhaustive]",
+    required_update: Minor,
+    lint_level: Allow,
+    reference_link: Some("https://doc.rust-lang.org/reference/attributes/type_system.html#the-non_exhaustive-attribute"),
+    query: r#"
+    {
+        CrateDiff {
+            current {
+                item {
+                    ... on Enum {
+                        visibility_limit @filter(op: "=", value: ["$public"]) @output
+                        name @output @tag
+
+                        attrs @filter(op: "not_contains", value: ["$non_exhaustive"])
+
+                        importable_path {
+                            path @output @tag
+                        }
+
+                        span_: span @optional {
+                            filename @output
+                            begin_line @output
+                        }
+                    }
+                }
+            }
+            baseline {
+                item {
+                    ... on Enum {
+                        visibility_limit @filter(op: "=", value: ["$public"])
+                        name @filter(op: "=", value: ["%name"])
+
+                        attrs @filter(op: "contains", value: ["$non_exhaustive"])
+                        
+                        importable_path {
+                            path @filter(op: "=", value: ["%path"])
+                        }
+                    }
+                }
+            }
+        }
+    }"#,
+    arguments: {
+        "public": "public",
+        "non_exhaustive": "#[non_exhaustive]",
+    },
+    error_message: "A Public API enum is no longer #[non_exhaustive]. External code may assume all variants are known, causing breaking changes if new ones are added.",
```

Comment:
Same here: please pay close attention to capitalization and style and ensure everything matches existing lints. For example, we don't capitalize `Public`.